docs/introduction/developer-quickstart.md

```diff
@@ -281,3 +281,77 @@ Notice how in the `visibilityRules` above, we configure the `DataChanged` event
+Note that it can be used for any dApp that requires a no-click UX.
+
+#### When the game starts
+
+Before the user can start playing, the game must create the SK and ask the user to move some funds to that address. 
+The funds will be used to pay for moves. 
+
+- Call the RPC ``sessionkeys_Create`` - without any parameters. This will return a hex-encoded address of the SK. 
+- Create a normal transaction that transfers some ETH to the SK. The amount depends on how many "moves" the user is prepared to prepay for. 
+- Ask the user to sign this transaction with their normal wallet, and submit it to the network using the library of your choice. 
+- Once the receipt is received, call ``sessionkeys_Activate``. 
+
+#### The game
+
+After activation of the SK, create a transaction for each move, but don't ask the user to sign them.
+Instead, submit them to the network unsigned using the RPCs: ``eth_sendRawTransaction`` or ``eth_sendTransaction``. 
+
+Because the SK is active, the platform will sign the transactions on behalf of the user.
+
+As a game developer, you are responsible to keep track of the balance of the SK. You can also query the network for the balance of the address.
+If the SK runs out of balance, you have to ask the user to move more funds to the SK.
+
+#### Finishing the game
+ 
+When a game ends, you have to move the remaining funds back to the main address and deactivate the key.
+
+- create a Tx that moves the funds back from the SK to the main address. Submit it unsigned, because the funds are controlled by the SK.
+- call the RPC: ``sessionkeys_Deactivate``- from now on, unsigned transactions will no longer be signed by the SK.
+
+
+## Game Security
+
+Every on-chain game developer knows that every move that relies on entropy must be executed in two steps. 
+
+Imagine you implement an on-chain coin flip game. The player pays 0.1ETH to choose `Heads` or `Tails`. 
+If they win, they receive 0.2ETH, otherwise they lose the 0.1ETH.
+Even if randomness is unpredictable, this simple game can be exploited in several ways:
+
+- The attacker can create a “proxy” smart contract to play on their behalf. Using a similar mechanism to flash loans in DeFi: the proxy is programmed to make multiple actions, and only “commit” if it can obtain a profit. In our case, if the coin flip is losing, the proxy can just revert. The only cost will be the gas burned. 
+- Transactions consume gas, and the gas cost can inadvertently reveal information. For instance, if a winning move is more computationally intensive than a losing one, players could deduce optimal moves by estimating gas costs for various actions.
+
+The typical solution is to use a commit-reveal scheme. The player commits to a move, and then reveals it. This way, the player can't change their mind after seeing the result.
+This solution has the major drawback that it introduces extra complexity, latency and cost.
```

Comment:
Games with partial reward also need to be careful; 

Player bets 1 eth, win -> 1.5 eth, lose -> refund 0.5 eth (lets ignore if the example makes sense in heads and tails) 
The callback done to determine if the player has won needs to pay in both cases; This can be griefed if the call to the address of the player reverts in case the reward is not what is expected. Then contracts usually have a deadline after which players can refund the money. The workaround for this is wrapping the reward in wETH and force giving it as the wETH contract won't care to revert .. for now.